src/Database/LSMTree/Internal/MergingRun.hs

```diff
@@ -200,7 +155,7 @@ new hfs hbio resolve caching alloc indexType mergeType runPaths inputRuns =
     withActionRegistry $ \reg -> do
       runs <- V.mapM (\r -> withRollback reg (dupRef r) releaseRef) inputRuns
       merge <- fromMaybe (error "newMerge: merges can not be empty")
-        <$> Merge.new hfs hbio caching alloc indexType (toMergeType mergeType) resolve runPaths runs
+        <$> Merge.new hfs hbio caching alloc indexType mergeType resolve runPaths runs
```

Comment:
The suggestion above would mean that `new` does still need the `IsMergeType t` constraint, but e.g. `supplyCredits` and the other functions would not need it.

Reply:
Now redundant.